src/components/App.tsx

```diff
@@ -7757,6 +7774,30 @@ class App extends React.Component<AppProps, AppState> {
     this.setState({ suggestedBindings });
   }
 
+  public setSelection(elements: readonly NonDeletedExcalidrawElement[]) {
+    const selectedElementIds: { [id: string]: true } = {};
+    const selectedGroupIds: { [id: string]: true } = {};
+
+    elements.forEach((ele) => {
+      if (ele.groupIds.length) {
+        selectedElementIds[ele.id] = true;
+        ele.groupIds.forEach((id) => {
+          selectedGroupIds[id] = true;
+        });
```

Comment:
as far as I know this is not correct as we don't select all groups if you're drilling into a parent group(s).

---

The whole selection thing is tricky, and even all this is not enough as there are frames and such. So either we need to rewrite this completely, or I'd just inline it for the mermaid's sake and leave this function for later.